polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -1075,7 +1060,37 @@ OffsetVector::getConstIntegerValue(size_t row, DataFlowSolver &solver) const {
 // MemoryAccess
 //===----------------------------------------------------------------------===//
 
+MemoryAccessMatrix
+MemoryAccess::getIntraThreadAccessMatrix(unsigned numThreads) const {
```

Comment:
Its the number of values produced by calls to `sycl.nd_item.get_global_ids`. The number of columns in the memory access matrix is equal to that value (leftmost columns) plus the number of induction variables in the loop nest that contains the memory access.  What term would you like ? I can change it.

Reply:
I think "dimensions" is the term used by the SYCL spec for this, so maybe `dimensions`? 

Just to explain my reasoning here: `nd_range<3>{{2,4,4}, ...}` has dimension `3`, but launches `32` work-items (i.e. threads). We are interested in the dimensionality `3` here, not the actual number of threads (potentially only known at compile time), so that's why I think `numThreads` might be confusing here.

Reply:
I think "dimensions" is the term used by the SYCL spec for this, so maybe `dimensions`? 

Just to explain my reasoning here: `nd_range<3>{{2,4,4}, ...}` has dimension `3`, but launches `32` work-items (i.e. threads). We are interested in the dimensionality `3` here, not the actual number of threads (potentially only known at compile time), so that's why I think `numThreads` might be confusing here.

Reply:
Yes lets use numGridDimensions.

Reply:
done